cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -98,6 +98,11 @@ public DatastoreManager(String directoryPath) {
      */
     public DatastoreManager(File directoryPath) {
         logger.fine("Datastore path: " + directoryPath);
+
+        if(!directoryPath.exists()){
+            directoryPath.mkdir();
```

Comment:
Do we know for sure that we'd only need to create 1 part of this path? i.e. should it use `mkdirs()` instead of `mkdir()`?

Also is it worth checking the boolean returned and throwing an `Exception` if the directory couldn't be created (for example insufficient permissions, or if using `mkdir` not `mkdirs` then it is possible the parent directory didn't exist)?